libs/components/core/src/lib/modules/sr-label/sr-label.directive.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  DestroyRef,
+  Directive,
+  ElementRef,
+  Input,
+  OnInit,
+  inject,
+} from '@angular/core';
+
+const SR_LABELS_CONTAINER_ID = 'sky-sr-labels-container';
+
+/**
+ * Adds the element to a screen reader only section of the body.
+ * This prevents components' DOM from including text only intended for screen readers.
+ */
+@Directive({
+  selector: '[skySrLabel]',
+  standalone: true,
+})
+export class SkySrLabelDirective implements OnInit {
+  readonly #elementRef = inject(ElementRef);
+  #_createLabel = true;
+
+  public ngOnInit(): void {
+    this.#insertSrLabel();
+  }
+
+  @Input()
+  public set destroyRef(ref: DestroyRef) {
+    ref.onDestroy(() => this.#removeSrLabel());
+  }
+
+  @Input()
+  public set createLabel(value: boolean | undefined) {
+    this.#_createLabel = value ?? true;
```

Comment:
Having the default value here be `true` when `undefined` is specified is a bit confusing if it's supposed to act like `*ngIf`. Passing `undefined` to `*ngIf` is equivalent to `false`, and it's not uncommon to use a falsy statement and not a boolean for `*ngIf` (e.g. `<div *ngIf="someNonBooleanObject && someOtherNonBooleanObject" />`). This might be a case where we need to use a negative input name with a default value of `false` or `undefined`, like `srHidden`.

Reply:
I'll change the name--I just think the default scenario would be always showing it, so I don't think we should have to specify to show it every time.

Reply:
The change I proposed would mean you'd only specify it when it's hidden.